client/state/initial-state.js

```diff
@@ -178,6 +179,10 @@ function getInitialPersistedState( initialReducer, currentUserId ) {
 			);
 
 			clearPersistedState();
+
+			// If state was randomly cleared save a flag indicating this in browser storage
+			const isStateRandomlyCleared = ! config.isEnabled( 'force-sympathy' );
+			setStoredItem( IS_STATE_RANDOMLY_CLEARED_KEY, isStateRandomlyCleared );
```

Comment:
@tyxla  I used `browser-storage` here instead of `persisted-state` because that seemed to be tailored/targeted for redux and query data.
https://github.com/Automattic/wp-calypso/blob/e7af7f07ed4a6466d2be0b5e5ccd31dc1b6d4aee/client/state/persisted-state.js#L17

So I directly used the browser-state abstraction.

Reply:
Oh, I didn't mean to persist that piece of data, but rather to have a simple variable as a flag that just lives inside that `persisted-state` module, because that way we can set that flag when we clear the state since the `clearPersistedState()` function lives there. What do you think about that?

Reply:
I feel like that approach is a bit hacky than the current approach. And it would be confusing to see a flag like that inside a module meant to handle persisted-state data. 

Do you think the current approach doesn't make sense? If crowding broswer storage is a problem I can actually remove the state once we inform the user. It would be like nothing happened.

Reply:
Could you elaborate on what makes it hacky from your perspective? Also, it's related to persisted state module itself, so I see a good reason to have it in that module, so maybe you could elaborate on that as well? I'm just seeking the simplest possible way to do this, and I don't see a simpler one than the flag one.

Reply:
I think this is also more easier to explore, from a traceability standpoint, for someone trying to understand the symphony dev notice feature. You just have to do a search for the storage key and you can instantly find the exact two places this state is accessed and set to understand what is happening.

If we go with the other approach the user will have to click across multiple times with getters and setters to figure out what is happening.

Reply:
FWIW the "where it lives" part is not so relevant for the discussion - we could move the flag to where the rest of the sympathy mode code lives and what I suggested would still stand, and I do feel what I suggest would be simpler, mostly because we don't need to clear the flag after we've displayed a notice (which is something necessary for your approach).

I don't have so strong feelings about it, but I do feel that using browser storage is still more complex than it should be for such a simple thing. The main reason is that we don't really leverage the browser storage in the way we usually do - as a persistence layer, but rather we use it as a runtime store, and I don't see the need for that. I might be missing something, though - do let me know if that's the case!

In any case, the current state of your proposal looks much better than the initial one, and I don't have many more thoughts to add.